src/mem/pagemap.h

```diff
@@ -241,7 +241,7 @@ namespace snmalloc
         (c->version != 1) || (c->is_flat_pagemap) ||
         (c->sizeof_pointer != sizeof(uintptr_t)) ||
         (c->pagemap_bits != GRANULARITY_BITS) ||
-        (c->size_of_entry != sizeof(T)) || (!std::is_integral_v<T>))
```

Comment:
You've deleted the is_integral check, but that probably means that we should be enforcing the type some other way.  Please extend the config struct to set `version = 2` for non-integral types and add a field at the end to specify that it's `void*` so that the checks still do the right thing and we can't mix pagemaps with incorrect compilation modes.

Reply:
OK; new version forthcoming.